src/main/java/appeng/fluids/container/ContainerFluidTerminal.java

```diff
@@ -355,20 +354,19 @@ public void doAction( EntityPlayerMP player, InventoryAction action, int slot, l
 			return;
 		}
 		ItemStack held = player.inventory.getItemStack();
-		if( !held.hasCapability( CapabilityFluidHandler.FLUID_HANDLER_ITEM_CAPABILITY, null ) )
-		{ // For now only do simple i/o with held tanks
+		if( held.getCount() != 1 )
+		{
+			// only support stacksize 1 for now
 			return;
 		}
 		IFluidHandlerItem fh = FluidUtil.getFluidHandler( held );
 		if( fh == null )
 		{
-			throw new NullPointerException( held.getDisplayName() + " did not give FLUID_HANDLER_ITEM_CAPABILITY" );
+			// only fluid handlers items
+			return;
 		}
-		boolean isBucket = held.getItem() == Items.BUCKET ||
-				held.getItem() == Items.WATER_BUCKET ||
-				held.getItem() == Items.LAVA_BUCKET ||
-				held.getItem() == Items.MILK_BUCKET ||
-				held.getItem() == ForgeModContainer.getInstance().universalBucket;
+		boolean isBucket = held.getItem() == Items.BUCKET || held.getItem() == Items.WATER_BUCKET || held.getItem() == Items.LAVA_BUCKET || held

```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1067)